src/lib/routes/client-api/feature.ts

```diff
@@ -199,12 +236,32 @@ export default class FeatureController extends Controller {
         req: IAuthRequest,
         res: Response<ClientFeaturesSchema>,
     ): Promise<void> {
+        if (this.flagResolver.isEnabled('optimal304')) {
+            return this.optimal304(req, res);
+        }
+
         const query = await this.resolveQuery(req);
 
         const [features, segments] = this.cache
             ? await this.cachedFeatures(query)
             : await this.resolveFeaturesAndSegments(query);
 
+        if (this.flagResolver.isEnabled('optimal304Differ')) {
```

Comment:
explain a bit what we expect to see here. 
Old cache is 600ms, while new cache should be faster.